timelock-impl/src/main/java/com/palantir/atlasdb/timelock/paxos/ClientAwarePaxosAcceptor.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.timelock.paxos;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import com.palantir.paxos.BooleanPaxosResponse;
+import com.palantir.paxos.PaxosPromise;
+import com.palantir.paxos.PaxosProposal;
+import com.palantir.paxos.PaxosProposalId;
+
+@Path("/" + PaxosTimeLockConstants.INTERNAL_NAMESPACE
+        + "/" + PaxosTimeLockConstants.CLIENT_PAXOS_NAMESPACE
+        + "/{client}"
+        + "/acceptor")
+public interface ClientAwarePaxosAcceptor {
```

Comment:
javadoc on why this exists and its use, possibly also package private if possible.